src/NServiceBus.Core/PropertyDictionary.cs

```diff
@@ -0,0 +1,173 @@
+﻿namespace NServiceBus
+{
+    using System.Collections;
+    using System.Collections.Generic;
+
+    sealed class PropertyDictionary : IReadOnlyDictionary<string, string>
```

Comment:
I wanted to make `EndpointInstance` a read-only and copying stuff from one dictionary to another seemed like a significant waste.